### DISCUSSION TITLE
Which dask scheduler for rioxarray?

### DISCUSSION THREAD

**Discussion by orttak** [corteva/rioxarray#237] [Q&A]
Hi, I am newbie user for rioxarray and dask and after some research I am bit confused about dask scheduler control. Should I change the scheduler for using rioxarray in production(django api )? When I use below code(direct_ndvi) , I got memory leaking error. According to dask documentation, I use  processes=False. When processes=True, everything okay. Main thing, I want to control memory usage when I work with multiple threads. I didn't see any example about single machine scheduler memory setting. Also, What is the default scheduler for rioxarray? Any tips to understand this issue? Thanks.

```

def direct_ndvi(red,nir):
 #rioxarray return
    with LocalCluster(n_workers=int(0.3 * mp.cpu_count()),
        processes=False,
        threads_per_worker=2,
        memory_limit='4GB',
        #ip='tcp://localhost:9895',
        ) as cluster, Client(cluster) as client:
            red_xarray=red
            nir_xarray=nir
            ndvi = (nir_xarray.astype(float) - red_xarray.astype(float))/(nir_xarray + red_xarray)
    return ndvi

```

**Comment by snowman2** [comment]
Here are examples for using dask with rioxarray:
- https://corteva.github.io/rioxarray/stable/examples/read-locks.html
- https://corteva.github.io/rioxarray/stable/examples/dask_read_write.html

For in-depth questions about dask, I recommend going to the dask developers for tips as they would likely give better answers.
